helpers/sortByMapping.js

```diff
@@ -9,32 +9,48 @@ module.exports = function( dust ) {
   * @description Sorting method that orders a list of things based on a provided map
   * @param {object} eventsObject Object to sort
   * @param {string} map Map to use to sort with
-  * @example {@_sortByMapping eventsObject=eventsReply.events map="ASHLKE, ASHWKE, ASHJBE} {/_sortByMapping} output loop of events
+  * @param {string} sortKeys - structure of response to sort on
+  * @example {@_sortByMapping sortObject=eventsReply.events sortKeys="links hotelProducts ids" map="PAUGLA, PAUFOR, PAUELM, PAUFAR"} {/_sortByMapping}
   */
 
   dust.helpers._sortByMapping = function( chunk, context, bodies, params ) {
-    var eventsObject = params.eventsObject;
-    var map = params.map.split(','); // Split string into array
+    // Object to sort
+    var sortObject = params.sortObject;
+    // Map to sort to
+    var map = params.map;
+    // Keys required for mapping
+    var keysParam = params.sortKeys.split(' ');
 
     var mappedObject = [];
 
-    // For every item in map array
+    var sortParameters = function( events, sortparams ) {
+      // Loop over sort parameter structure passed in tpl
+      for ( var i = 0; i < sortparams.length; i++ ) {
+        if ( !events ) return null;
+        events = events[sortparams[i]];
+      }
+      return events;
+    };
+    // For every item in map array passed
     _.forEach( map, function( id ) {
-      // For every event if the id = map id then add to mappedObject
-      _.forEach( eventsObject, function( event ) {
-        if ( event.id === id ) {
-          mappedObject.push( event );
+      // Loop over the object to be sorted
+      _.forEach( sortObject, function( matchObject ) {
+        var requiredProduct = sortParameters( matchObject, keysParam );
```

Comment:
I wouldn't call this `xxxProduct`, there's no guarantee this function will be used to sort "products"